apps/mobile/src/features/approver/utils.tsx

```diff
@@ -124,19 +129,46 @@ export function getFormReviewTxSummary({ tx, symbol }: ReviewTxSummaryProps) {
       spendingCondition: { fee },
     },
   } = tx;
-  if (payload.payloadType !== PayloadType.TokenTransfer) {
-    throw new Error('Unsupported payload type');
+  if (payload.payloadType === PayloadType.TokenTransfer) {
+    const { amount: txValue, recipient } = payload;
```

Comment:
This fn seems to be responsible for parsing all these different tx types. Why not have them as separate methods? Also, What does `formReview` have to do with this? I don't see a form. Not does it have any form-specific behaviour. It's just effectively deserialising the tx a certain way.